src/DotnetRelease/PatchReleaseOverviewRecords.cs

```diff
@@ -0,0 +1,161 @@
+using System.ComponentModel;
+using System.Text.Json.Serialization;
+
+namespace DotnetRelease;
+
+public record PatchReleaseOverview(
+    [property: Description("Major (or major.minor) version of product.")]
+    string ChannelVersion,
+
+    [property: Description("The date of the patch release.")]
+    DateOnly ReleaseDate,
+
+    [property: Description("The version (branding) of the release")]
+    string ReleaseVersion,
+
+    [property: Description("Whether the release contains any CVE fixes.")]
+    bool Security,
+
+    [property: Description("A patch release with detailed release information.")]
+    PatchRelease Release);
+
+public record PatchRelease(
+    [property: Description("The date of the patch release.")]
+    DateOnly ReleaseDate,
+
+    [property: Description("The version (branding) of the release")]
+    string ReleaseVersion,
+
+    [property: Description("Whether the release contains any CVE fixes.")]
+    bool Security,
+
+    [property: Description("The CVEs disclosed with the release.")]
+    IList<Cve> CveList,
+
+    [property: Description("A URL to release notes.")]
+    string ReleaseNotes,
+
+    [property: Description("Runtime component of the release.")]
+    RuntimeComponent Runtime,
+
+    [property: Description("SDK component of the release (primary SDK release).")]
+    SdkComponent Sdk,
+
+    [property: Description("SDK components of the release (often multiple, otherwise a repeat of `Sdk` value within an array).")]
+    IList<SdkComponent> Sdks,
+
+    [property: Description("ASP.NET Core component of the release"),
+        JsonPropertyName("aspnetcore-runtime")]
+    AspNetCoreComponent AspNetCoreRuntime,
+
+    [property: Description("Windows Desktop component of the release."),
+        JsonPropertyName("windowsdesktop")]
+    Component? WindowsDesktop = null);
+
+[Description("A disclosed vulnerability (AKA CVE).")]
+public record Cve(
+    [property: Description("The ID tracking the CVE.")]
+    string CveId,
+    
+    [property: Description("The URL tracking the CVE at the authoritative site.")]
+    string CveUrl);
+
+[Description("Runtime component of a release.")]
+public record RuntimeComponent(
+    [property: Description("The version of the component.")]
+    string Version,
+
+    [property: Description("The display or branding variant of the version, if different.")]
+    string VersionDisplay,
+
+    [property: Description("The version of Visual Studio that includes this component version."),
+        JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    string VSVersion,
```

Comment:
Is `VSVersion` correctly mapped to `vs-version` or would it need to be `VsVersion` to not end up `v-s-version`? Same for some other VS properties.

Reply:
I assume you can't test it with that file, since on reading empty ones, this would have the same result as if not matched (as it's `string` and not `string?`).

Reply:
How about here?

https://github.com/dotnet/core/blob/7f92191682fd42cf601809f7a7105c3938e54813/release-notes/8.0/8.0.7/release.json#L28

This code path is testing it, right?

https://github.com/richlander/distroessed/blob/f79a42b4ed15a797e4f27cc95443d052a7fd4537/src/UpdateReleasesIndex/Program.cs#L53